dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java

```diff
@@ -587,6 +603,77 @@ public Object preInvoke(HttpServletRequest req, HttpServletResponse resp, String
         return webSecurityContext;
     }
 
+    /**
+     * Collect all the Http header names and values.
+     *
+     * @param req HttpServletRequest
+     * @return Returns a string that contains each parameter and it value(s)
+     *         in the HttpServletRequest object.
+     */
+    @Trivial
+    private String servletRequestInfo(HttpServletRequest req) {
+
+        StringBuffer sb = new StringBuffer(DEBUG_REQ_INFO_BUFSIZE);
+        sb.append(DEBUG_HEADER_EYECATCHER).append("\n");
+
+        try {
+            Enumeration<String> headerNames = req.getHeaderNames();
+            while (headerNames.hasMoreElements()) {
+                String headerName = headerNames.nextElement();
+                sb.append(headerName).append("=");
+                sb.append("[").append(getHeader(req, headerName)).append("]\n");
+            }
+        } catch (Throwable t) {
+            // do nothing because it probably means the parser was trying to parse
+            // non form data or serialized object data.  This is a trace issue and
+            // has nothing to do with the spec.
+        }
+
+        if (req.getContextPath() != null)
+            sb.append(DEBUG_CONTEXT_PATH).append(req.getContextPath());
+        if (req.getServletPath() != null)
+            sb.append(DEBUG_SERVLET_PATH).append(req.getServletPath());
+        if (req.getPathInfo() != null)
+            sb.append(DEBUG_PATH_INFO).append(req.getPathInfo());
+        return sb.toString();
+    }
+
+    /**
+     * @r5
```

Comment:
Should the documentation on this method description be re-evaluated?

Reply:
Yes,  I will update it. That's good comment.  Now I realize  I copy and pasted tWAS method.   I will work on it.